### PR TITLE
Implementation of Trick Leader.

### PR DESCRIPTION
**Pull request by davidjha**
Please take a look, I modified trick leader. It works however it will not be able to determine trick leader after the 4th player has played. At which point do we need to know who the trick leader is? Perhaps we go with adding the additional encoding (31-34) to our state for trick leader.

### CODE REVIEW

**Comment by irobert4** [comment]
David, how do you know that it works?  Did you do any testing on it?

I think we kind of have two separate issues right now. A way that we can determine the trick leader is by subtracting from the current player (since the players are going clockwise), but to do any sort of trick-leader-searching properly we need to first start by actually setting the trick leader correctly (through a trick winner function maybe? the player with the highest card following suit wins)

To set the trick winner/trick leader properly the only way I can think of to do this is through an encoding of 31-34. 

Basically the code to find the trick leader and actually set the trick leader are separate. We need to first decide how we want to set the trick leader.

David and I are for an encoding, it looks like.  

@aphelps4 @msomarri @kruzaavn @c-to-the-fazzy 
What do you guys think?

**Comment by irobert4** [comment]
Also I just thought of another issue.  How are we going to differentiate between the trick winner of the previous round and trick winners of older rounds?

Whoever won the trick is the current player of the next round, that somehow needs to be indicated to the adjudicator. If we use the same encoding for trick winners of the previous round and trick winners of older rounds, how do we differentiate them so the correct trick winner is chosen as the first player of the next round?

**Comment by davidjha** [comment]
I think this is resolved with Isabella's  Isabellas-Cleanup-Branch-Review #42. Using the new encodings for trick leader, trick leader can be inferred from the state quite easily.
